### ISSUE TITLE
Configuration Options

### ISSUE THREAD

**Issue by justinobney** [ee92/folderize#3]
The first examples that come to mind are:

- test file extension (I use `foo.spec.ts`, default to `test`)
- test subdirectory (I put under `./__tests__/foo.spec.ts`, default to `./`)

**Comment by ee92** [comment]
@justinobney I agree some config is needed. Ideally the 'additional files' picker would be completely customizable. Still not sure exactly what that would look like, but I'll play around with it. In the meantime feel free to make PR 😃

**Comment by floede** [comment]
May I suggest the option to add .md files?
I use that all the time for Styleguidist.

Also the option to choose .scss or .less for stylesheets.

And of course, some kind of config to define file name patterns :-)

**Comment by ppluchon** [comment]
Hello @ee92 ! 

I am a huge fan of your extension, congratz for the awesome work ! 

Do you think you can add those enhancements ? 

Thanks a lot for your feedback.

**Comment by agch-dev** [comment]
Hello, @ee92 awesome extension! I stumbled upon this same problem. I Decided to try and implement it to give vscode extensions and typescript a try.

I have a working prototype that I need to polish before submitting a PR but wanted to share my progress here to let you know I am working on it.

![](http://g.recordit.co/681EmPwXxS.gif)

And in my vscode settings I have:
![imagen](https://user-images.githubusercontent.com/9297073/90335001-059e9480-dfa8-11ea-8945-b38ab7073743.png)

I still have to work on adding the correct imports to the files and think about how that might look like on a user-added option. As well as adding some documentation on how this preset, custom options system should be used by users in the settings.

A quick question that I have is why is `createFile` optional in the `Option` type? is there any value in having an option that does not create any file?

**Comment by agch-dev** [comment]
For people interested in more customization options since this issue is kinda dead I found this neat extension that lets you do exactly what we want and more https://github.com/Huuums/vscode-folder-templates
